DOC-9342: Limitations on key-length changed in 7.0 - docs incorrect

RayOffiah opened the pull request:
Changed the size of the key from 250 bytes to 246 bytes
(unless it's the default collection)

RayOffiah committed:
DOC-9342:
Changed the size of the key from 250 bytes to 246 bytes
(unless it's the default collection)

RayOffiah committed:
DOC-9391:
Changed the size of the key from 250 bytes to 246 bytes
(unless it's the default collection)

jimwwalker commented on modules/learn/pages/data/data.adoc:
```diff
@@ -24,8 +24,11 @@ Each key:
-* Must be unique within its bucket.
+* May be no longer than 246 bytes.
++
+NOTE: For compatibility with earlier versions of Couchbase, the key string can be 250 bytes for the default collection.
```
It needs to be made clearer what that compatibility means to the user.  In this case, if they are accessing the default collection 250 bytes is the maximum, they could be accessing the default collection because of a 'legacy' SDK or they could just be explicitly using that particular collection.

simon-dew reviewed (commented):
Please re-request review when @jimwwalker 's comments addressed

RayOffiah committed:
DOC-9342:
Added note about accessing through legacy SDK

RayOffiah committed:
Merge remote-tracking branch 'origin/DOC-9342-Limitations_on_key-length_changed_in_7.0_-_docs_incorrect' into DOC-9342-Limitations_on_key-length_changed_in_7.0_-_docs_incorrect

# Conflicts:
#	modules/learn/assets/images/data/item-maximum-sizes.png
#	modules/learn/pages/data/data.adoc

RayOffiah commented:
Hi, @jimwalker. This is ready to review again when you have a minute.

simon-dew commented on modules/learn/pages/data/data.adoc:
```diff
@@ -24,8 +24,15 @@ Each key:
++
+[NOTE]
+====
+If a user accesses the *default* collection using a legacy SDK, then the key can be up to 250 bytes in the length.
```
If I understood @jimwwalker 's comments correctly, it doesn't have to be because they're using a legacy SDK, that's just one example.

```suggestion
If a user accesses the *default* collection, for example by setting the query context to the default collection in the default scope, or by using a legacy SDK, then the key can be up to 250 bytes in length.
```

RayOffiah replied:
Yup, that makes sense.

simon-dew reviewed (changes requested):
Sorry, I don't think it's quite there. Suggested a change

RayOffiah committed:
Update modules/learn/pages/data/data.adoc

Co-authored-by: Simon Dew <39966290+simon-dew@users.noreply.github.com>

RayOffiah committed:
DOC-9342:
Adjustment to image to make it clear that accessing we're using the legacy SDK as an example

simon-dew reviewed (approved):
LGTM 👍
